### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by itsmraga-hub**
# 💯 Achievements

- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 <h1>Users#index</h1>
-<p>Find me in app/views/users/index.html.erb</p>
+<ul>
+  <% @users.all.each do |user| %>
+    Name: <%= user.name %>
+    Number_of_posts: <%= user.posts_counter %>
+  <% end %>
+</ul>
```
Comment (TedLivist) [inline app/views/users/index.html.erb]: - [x] Each user name should be a link that leads to the user's show page.

- [x] Consider improving your design to match the wireframe for the user index page.

  Your Implementation | Wireframe
  :---:|:---:
  user-2 | user-1

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,20 @@
...
+    <% @latest_three_posts.each do |post| %>
+      <li>
+        <h3><%= post.title %></h3>
+        <p><%= post.text %></p>
+        <p><%= post.comments_counter %></p>
+        <p><%= post.likes_counter %></p>
+      </li>
+    <% end %>
+  </ul>
+</div>
```
Comment (TedLivist) [inline app/views/users/show.html.erb]: - [ ] Kindly fix your user show page design and layout to match the wireframe as well

  Your Implementation | Wireframe
  :---:|:---:
  show-2 | show-1

- [ ] The **See all posts** button should lead to the user's posts index page

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,16 @@
...
+    <% @user_posts.each_with_index do |post, i| %>
+      <li>
+        <h3><span><%= i + 1 %></span> <%= post.title %></h3>
+        <p><%= post.text %></p>
+        <p>Comments: <%= post.comments_counter %></p>
+        <p>Likes: <%= post.likes_counter %></p>
+      </li>
+    <% end %>
+  </ul>
+</div>
```
Comment (TedLivist) [inline app/views/posts/index.html.erb]: - [ ] Kindly ensure your posts index page matched the wireframe.

  Your Implementation | Wireframe
  :---:|:---:
  posts-1 | posts-2

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,16 @@
...
+  <ul>
+    <% @user_posts.each_with_index do |post, i| %>
+      <li>
+        <h3><span><%= i + 1 %></span> <%= post.title %></h3>
+        <p><%= post.text %></p>
+        <p>Comments: <%= post.comments_counter %></p>
+        <p>Likes: <%= post.likes_counter %></p>
+      </li>
+    <% end %>
+  </ul>
```
Comment (TedLivist) [inline app/views/posts/index.html.erb]: - [ ] Based on the wireframe, after listing the post details you also need to list the recent comments belonging to the post, using the **recent_five_comments** method in your Post model.

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1 +1,9 @@
...
\ No newline at end of file
+<div>
+  <h2>Post #<span><%= @post.id %></span> by <span><%= @user.name %></span></h2>
+  <p><%= @post.text %></p>
+  <ul>
+    <% @post_comments.each do |comment| %>
+      <li><span><%= comment.author.name%> </span><%= comment.text %></li>
+    <% end %>
+  </ul>
+</div>
```
Comment (TedLivist) [inline app/views/posts/show.html.erb]: - [ ] Kindly design your post show page according to the wireframe.

  Your Implementation | Wireframe
  :---:|:---:
  jui | jui-2

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @itsmraga-hub,

Good job so far!
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (itsmraga-hub) [reply app/views/users/index.html.erb]: Okay. I'll work on the changes

File: app/views/posts/show.html.erb [hunk 6]
```diff
@@ -1 +1,15 @@
-<h1>Post 1</h1>
\ No newline at end of file
+<section class="posts-show">
+  <div class="post-header">
+    <h2>Post #<span><%= @post.id %></span> by <span><%= @user.name %></span></h2>
```
Comment (rustam-tolipov) [inline app/views/posts/show.html.erb:3]: _You did an excellent job, however, there are small places you need to work on._ 

- [ ] To make your code professional, Kindly remove commented code in this line

File: app/views/posts/index.html.erb [hunk 7]
```diff
@@ -1 +1,36 @@
...
+  <ul>
+    <%= render @user %>
+  </ul>
+</section>
+<section class="">
+  <ul class="posts_list">
+    <% @user.posts.each_with_index do |post, i| %>
+      <li class="single_post">
+        <div class="post-header">
+          <h3>Post #<span><%= i + 1 %> </span>
```
Comment (rustam-tolipov) [inline app/views/posts/index.html.erb:11]: - [ ] To make your code professional, Kindly remove commented code in this line :)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @itsmraga-hub,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
One optional suggestion to make it approve could you remove this line from your readme?

Screenshot 2022-12-01 at 12 54 34

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 6 above)
Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: That is not commented code @Rustamxon7

File: app/views/posts/index.html.erb (hunk 7 above)
Comment (itsmraga-hub) [reply app/views/posts/index.html.erb:11]: Hi @Rustamxon7  this is not commented code. If you checked the view it displays

`Post # 1`

File: app/views/posts/show.html.erb (hunk 6 above)
Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: @Rustamxon7 please reply, should I resubmit. I have already used up two code reviews and this was unfairly graded

Comment (rustam-tolipov) [reply app/views/posts/show.html.erb:3]: Hello, just a sec...

Comment (rustam-tolipov) [reply app/views/posts/show.html.erb:3]: Screenshot 2022-12-01 at 11 58 54

In my vscode this is shown as commented.

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: Maybe just an extension man

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: If you check the render, you can see the post.id and the @user.name

Comments in HTML normally start with <\!--comment-->

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: Why would I comment out the closing <\/h2> tag? Do I resubmit?

Comment (rustam-tolipov) [reply app/views/posts/show.html.erb:3]: Yes, I agree with you. However, this is the guideline we need to follow. I'm reaching out to Senior Code reviewers. To get the full answer. Hope you will find some minutes :).

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: code3

@Rustamxon7

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: @Rustamxon7 please advise

Comment (rustam-tolipov) [reply app/views/posts/show.html.erb:3]: > If you check the render, you can see the post.id and the @user.name
> 
> Comments in HTML normally start with 

In HTML you are right, but in the .erb file, it accepts it as a comment.

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: But you can see clearly in my vscode its not a comment

code4

This is my index.html.erb file man

Comment (itsmraga-hub) [reply app/views/posts/show.html.erb:3]: @Rustamxon7 I just added line no. 4; you can see it's highlighted as a comment. You are using an extension in your vscode that apparently takes anything starting with a # symbol as a comment. Hash \# symbols are used as comments in the ruby language and python.

Comment (rustam-tolipov) [reply app/views/posts/show.html.erb:3]: @itsmraga-hub, I have reached out to Seniors, and they accepted your request. Your code is totally fine. You are able to re-request another review without changing the (#)gs.

**Review by rustam-tolipov — APPROVED** [review]
Hi @itsmraga-hub, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by itsmraga-hub** [comment]
Thanks. Cheers 🥂  to learning and coding. Good day.

**Pull request merged**